src/binding/vm.rs

```diff
@@ -20,3 +20,67 @@ pub fn require(name: &str) {
         vm::rb_require(name.as_ptr());
     }
 }
+
+
+pub fn thread_call_without_gvl<F, G>(func: F, unblock_func: Option<G>)
+    where F: FnMut(),
+          G: FnMut()
+{
+    unsafe {
+        vm::rb_thread_call_without_gvl(callbox as CallbackPtr,
+                                       Box::into_raw(Box::new(Box::new(func) as Box<FnMut()>)) as *const c_void,
```

Comment:
Extract `Box::into_raw(Box::new(Box::new(...) as Box<FnMut()>)) as *const c_void` into a separate function to DRY up the code (as I see, it is used several times here).

You can put it to `src/util.rs` which contains several functions for type casting.